src/unsigned.rs

```diff
@@ -883,78 +890,216 @@ macro_rules! uint_impl {
+                    self.value().count_zeros().wrapping_sub(Self::UNUSED_BITS as u32)
                 }
 
-                /// Returns the number of leading ones in the binary representation of self.
+                /// Returns the number of leading ones in the binary representation of `self`.
+                ///
+                /// # Examples
+                ///
+                /// Basic usage:
+                ///
+                /// ```
+                /// # use arbitrary_int::{u7, Number};
+                /// let n = !(u7::MAX >> 2);
+                /// assert_eq!(n.leading_ones(), 2);
+                ///
+                /// let zero = u7::new(0);
+                /// assert_eq!(zero.leading_ones(), 0);
+                ///
+                /// let max = u7::MAX;
+                /// assert_eq!(max.leading_ones(), 7);
+                /// ```
                 #[inline]
                 pub const fn leading_ones(self) -> u32 {
-                    let shift = ((core::mem::size_of::<$type>() << 3) - BITS) as u32;
-                    (self.value << shift).leading_ones()
+                    (self.value() << Self::UNUSED_BITS).leading_ones()
                 }
 
-                /// Returns the number of leading zeros in the binary representation of self.
+                /// Returns the number of leading zeros in the binary representation of `self`.
+                ///
+                /// # Examples
+                ///
+                /// Basic usage:
+                ///
+                /// ```
+                /// # use arbitrary_int::{u7, Number};
+                /// let n = u7::MAX >> 2;
+                /// assert_eq!(n.leading_zeros(), 2);
+                ///
+                /// let zero = u7::new(0);
+                /// assert_eq!(zero.leading_zeros(), 7);
+                ///
+                /// let max = u7::MAX;
+                /// assert_eq!(max.leading_zeros(), 0);
+                /// ```
                 #[inline]
                 pub const fn leading_zeros(self) -> u32 {
-                    let shift = ((core::mem::size_of::<$type>() << 3) - BITS) as u32;
-                    (self.value << shift).leading_zeros()
+                    if Self::UNUSED_BITS == 0 {
+                        self.value().leading_zeros()
+                    } else {
+                        // Prevent an all-zero value reporting the underlying type's entire bit width by setting
+                        // the first unused bit to one, causing `leading_zeros()` to ignore the unused bits.
+                        let first_unused_bit_set = const { 1 << (Self::UNUSED_BITS - 1) };
+                        ((self.value() << Self::UNUSED_BITS) | first_unused_bit_set).leading_zeros()
+                    }
                 }
 
-                /// Returns the number of leading ones in the binary representation of self.
+                /// Returns the number of trailing ones in the binary representation of `self`.
```

Comment:
whoops :)

Reply:
It's not exactly the most impactful change, but I figured I might as well :)